tests/performance/apdex-score/settings-provider.js

```diff
@@ -0,0 +1,83 @@
+const { resolve, extname } = require('node:path');
+const { existsSync, readFileSync } = require('node:fs');
+
+let SETTINGS_CACHE;
+
+const getInstanceURL = () => {
+  return SETTINGS_CACHE?.instanceURL;
+};
+
+const getIterations = () => {
+  return SETTINGS_CACHE?.iterations;
+};
+
+const getTestsToSkip = () => {
+  return SETTINGS_CACHE?.skip;
+};
+
+const hasPrivacyPolicy = () => {
+  return !!SETTINGS_CACHE?.hasPrivacyPolicy;
+};
+
+const getCapabilitiesSettings = () => {
+  return SETTINGS_CACHE?.capabilities;
+};
+
+const getUser = (type, role) => {
+  return SETTINGS_CACHE?.users?.find(user => user.type === type && user.role === role);
+};
+
+const getPage = (name) => {
+  return SETTINGS_CACHE?.pages && SETTINGS_CACHE.pages[name];
+};
+
+const getForm = (name) => {
+  return SETTINGS_CACHE?.forms && SETTINGS_CACHE.forms[name];
+};
+
+const getCommonElements = () => {
+  return SETTINGS_CACHE?.commonElements;
+};
+
+const settingsProvider = {
+  getIterations,
+  getInstanceURL,
+  getTestsToSkip,
+  hasPrivacyPolicy,
+  getCapabilitiesSettings,
+  getUser,
+  getPage,
+  getForm,
+  getCommonElements,
+};
+
+const loadSettings = () => {
+  if (SETTINGS_CACHE) {
+    return settingsProvider;
```

Comment:
I think we should figure out a way for classes to require this instead of passing it as a parameter everywhere. Is that possible?

Reply:
@dianabarsan I'm not sure I clearly understand what you mean here - can you explain further?

Reply:
In files like [contacts.page.js](https://github.com/medic/cht-core/pull/9037/files#diff-99ad0d385201276b69179d173fd2d2a42a9fd101bdf07d2421f9cc7825e171dd) every function receives this object as a parameter. It's super bloated. I'm suggesting we find a way to instantiate this somehow in the class instead of passing it as a parameter to every function. This applies to all files, not just the one gave as example.

Reply:
@dianabarsan This is what it looks like when we try to instantiate the function within the class:
```
const Page = require('@page-objects/apdex/page');
const loadSettings = require('../../performance/apdex-score/settings-provider');
const CONTACT_LIST = 'contactList';
const settingsProvider = loadSettings();

class ContactsPage extends Page {

  async loadContactList() {
    await super.loadPage(settingsProvider, CONTACT_LIST);
  }
}
```

I'm not entirely sure about the benefit of doing this because after doing this in all the 8 page files we still have to instantiate this same object in [test.spec.js](tests/performance/apdex-score/specs/test.spec.js) in order to skip the tests and also to define the number of iterations - won't it be better to have this instantiated in only the test file?

Reply:
Can we instantiate it in `@page-objects/apdex/page` ?

Reply:
My intent is to never pass the settings as a parameter at all

Reply:
@dianabarsan the requested changes have now been made - kindly have a look again
Thanks